types/block_test.go

```diff
@@ -1389,8 +1390,8 @@ func TestPutBlock(t *testing.T) {
 
 func generateRandomData(msgCount int) Data {
 	out := make([]Message, msgCount)
-	for i, msg := range generateRandNamespacedRawData(msgCount, NamespaceSize, ShareSize) {
-		out[i] = Message{NamespaceID: msg[:NamespaceSize], Data: msg[:NamespaceSize]}
+	for i, msg := range generateRandNamespacedRawData(msgCount, NamespaceSize, MsgShareSize-2) {
+		out[i] = Message{NamespaceID: msg[:NamespaceSize], Data: msg[NamespaceSize:]}
```

Comment:
the extra `-2` here is so that the msg shares don't unexpectedly split when adding the length delimiter.

Reply:
I agree, we should definitely test with larger message sizes, random txs, and ISRs. The issue I ran into is that we need to have a power of two square size for the ipld plugin to use generate Cids as expected (ref #257), and we don't actually have a way to know how many shares a message takes up until after the shares are computed ref #77.

I've changed the name of `generateRandomData` to `generateRandomMsgOnlyData` to hopefully make things slightly clearer, but it does not address the underlying issue. https://github.com/lazyledger/lazyledger-core/pull/235/commits/64e5bfd08d0d6f9749756bab0acef9ee99bb7a55

Reply:
just as an update, Ismail and I figured out the issues, and it was the last copy-append bug. Everything works as expected now, and we are planning on adding robust testing for computing shares from block data later when completing #257